File: backend/src/main/java/com/project2/domain/post/validation/ImageFile.java
package com.project2.domain.post.validation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import jakarta.validation.Constraint;
import jakarta.validation.Payload;

@Constraint(validatedBy = ImageFileValidator.class)
@Target({ElementType.FIELD})
@Retention(RetentionPolicy.RUNTIME)
public @interface ImageFile {
	String message() default "Only image files are allowed (jpg, jpeg, png, gif, webp)";

	Class<?>[] groups() default {};

	Class<? extends Payload>[] payload() default {};
}

File: backend/src/main/kotlin/com/project3/domain/post/validation/ImageFile.kt
package com.project3.domain.post.validation

import jakarta.validation.Constraint
import jakarta.validation.Payload
import kotlin.reflect.KClass

@MustBeDocumented
@Constraint(validatedBy = [ImageFileValidator::class])
@Target(AnnotationTarget.FIELD)
@Retention(AnnotationRetention.RUNTIME)
annotation class ImageFile(
        val message: String = "Only image files are allowed (jpg, jpeg, png, gif, webp)",
        val groups: Array<KClass<*>> = [],
        val payload: Array<KClass<out Payload>> = []
)